File: SWE4302_OOP_2/Lab_03/VehicleMS/src/VehicleManager.java
public class VehicleManager {
    public void StartVehicle(Vehicle vehicle) {
        vehicle.Start();
    }

    public void StopVehicle(Vehicle vehicle) {
        vehicle.Stop();
    }

    public void AccelerateVehicle(Vehicle vehicle) {
        if (vehicle instanceof IDriveable) {
            ((IDriveable) vehicle).AccelerateEngine();
        }
    }

    public void RideVehicle(Vehicle vehicle) {
        if (vehicle instanceof IRideable) {
            ((IRideable) vehicle).Ride();
        }
    }

}

File: SWE4302_OOP_2/Lab_03/C_Sharp_VehicleMS/VehicleMS/VehicleManager.cs
namespace VehicleMS
{
    public class VehicleManager
    {
        public void StartVehicle(Vehicle vehicle)
        {
            vehicle.Start();
        }

        public void StopVehicle(Vehicle vehicle)
        {
            vehicle.Stop();
        }

        public void AccelerateVehicle(Vehicle vehicle)
        {
            if (vehicle is IDriveable driveable)
            {
                driveable.AccelerateEngine();
            }
        }

        public void RideVehicle(Vehicle vehicle)
        {
            if (vehicle is IRideable rideable)
            {
                rideable.Ride();
            }
        }
    }

}
